sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/convert/MappingCosmosConverter.java

```diff
@@ -143,11 +159,66 @@ public JsonNode writeJsonNode(Object sourceEntity) {
             cosmosObjectNode.put("id", id);
         }
 
+        if (stripTransientFields) {
+            @SuppressWarnings("unchecked") final Class<T> domainType = (Class<T>) sourceEntity.getClass();
+            @SuppressWarnings("unchecked")
+            CosmosEntityInformation<T, Object> entityInfo = (CosmosEntityInformation<T, Object>) CosmosEntityInformation.getInstance(domainType);
+            List<String> transientFields =  entityInfo.getTransientFields();
+
+            if (!transientFields.isEmpty()) {
+                //strip fields with @Transient annotation from the JsonNode so they are not persisted.
+                cosmosObjectNode = stripTransientFields(cosmosObjectNode, transientFields);
+            }
+        }
+
         mapVersionFieldToEtag(sourceEntity, cosmosObjectNode);
 
         return cosmosObjectNode;
     }
 
+    /**
+     * To repopulate any transient fields that were stripped from the original object
+     * @param <T> type of source entity
+     * @param originalObjectToSave must not be {@literal null}
+     * @param responseItem must not be {@literal null}
+     * @return CosmosItemProperties
+     * @throws MappingException no mapping metadata for entity type
+     * @throws CosmosAccessException fail to map document value
+     */
+
+    public <T> JsonNode repopulateAnyTransientFields(Object originalObjectToSave, JsonNode responseItem) {
```

Comment:
We shouldn't need this function anymore, we can just call writeJsonNode with stripTransientFields as true to get the stripped version and call it again with stripTransientFields as false to get it with the transient fields still there to return.

Reply:
Actually I see where we can't just call writeJsonNode with skipTranisentFields as false because we are doing it on the response item, which won't have the transient fields. However I think we should go with the logic I had proposed to avoid the deepCopy(). Thoughts @kushagraThapar ?

Reply:
We do need it because the response item that comes back contains system fields (etag etc) from database, but has all the transient fields with null values. We need the values from the original object, and the system values from the response object.

Reply:
> Actually I see where we can't just call writeJsonNode with skipTranisentFields as false because we are doing it on the response item, which won't have the transient fields. However I think we should go with the logic I had proposed to avoid the deepCopy(). Thoughts @kushagraThapar ?

The deepCopy() is not needed even in the method as it is, removing it and cleaning up...

Reply:
> We shouldn't need this function anymore, we can just call writeJsonNode with stripTransientFields as true to get the stripped version and call it again with stripTransientFields as false to get it with the transient fields still there to return.
> 
> We want to try to remove all this logic, the deepcopy and everything here is very expensive and not going to be ideal.

Ok, so I've refactored this so mapping is only ever executed when needed. That is, when there are `@Transient` fields that actually have values set. Given that we have to do it, I think it is better to do it with JsonNode than hardcode the handling of every data type in ObjectNode.